src/components/ItemsList.js

```diff
@@ -1,38 +1,77 @@
 import React from 'react';
 import firebase from '../lib/firebase';
 import Nav from './Nav';
+import '../styles/ItemsList.css';
 import { useCollectionData } from 'react-firebase-hooks/firestore';
 
 const db = firebase.firestore().collection('shopping_list');
 
+const wasItemPurchasedWithinLastOneDay = (lastPurchasedOn) => {
+  const oneDayInMilliseconds = 24 * 60 * 60 * 1000;
+  return Date.now() - lastPurchasedOn <= oneDayInMilliseconds;
+};
+
 const ItemsList = () => {
   const userToken = localStorage.getItem('token');
 
   const [shoppingList, loading, error] = useCollectionData(
     db.where('token', '==', userToken),
+    { idField: 'documentId' },
   );
 
+  const markItemAsPurchased = (index) => {
+    const { items, documentId } = shoppingList[0];
+    const shoppingListObject = items[index];
+
+    if (shoppingListObject.lastPurchasedOn === null) {
+      shoppingListObject.lastPurchasedOn = Date.now();
+    } else {
+      shoppingListObject.lastPurchasedOn = null;
+    }
+
+    items[index] = shoppingListObject;
+    db.doc(documentId)
+      .update({
+        items: items,
+      })
+      .then(() => console.log('Successfully updated item'))
+      .catch((e) => console.log('error', e));
+  };
   return (
-    <div>
+    <div className="shoppinglist">
       <h1>Your Shopping List</h1>
-      <div>
-        {loading && <p>Loading...</p>}
-        {error && <p>An error has occured...</p>}
-        {shoppingList && !shoppingList.length && (
-          <p>You haven't created a shopping list yet...</p>
-        )}
+      {loading && <p>Loading...</p>}
+      {error && <p>An error has occured...</p>}
+      {shoppingList && !shoppingList.length && (
+        <p>You haven't created a shopping list yet...</p>
+      )}
+      <form>
         <ul>
           {shoppingList &&
             shoppingList[0] &&
             shoppingList[0].items.map((shoppingItemObject, index) => {
               return (
                 <li key={shoppingItemObject.shoppingListItemName + index}>
-                  {shoppingItemObject.shoppingListItemName}
+                  <input
+                    type="checkbox"
+                    id={shoppingItemObject.shoppingListItemName}
+                    onChange={() => markItemAsPurchased(index)}
+                    checked={
+                      shoppingItemObject.lastPurchasedOn === null
```

Comment:
I love how you check for if an item has been purchased within the last day, and feed this into the checked status of the checkbox! This is great. A small suggestion, but I wonder if it would be a bit simpler to move the check for null up into the wasItemPurchasedWithinLastOneDay function, so here it would be `checked={wasItemPurchasedWithinLastOneDay(shoppingItemObject.lastPurchasedOn)}` and then the the wasItemPurchasedWithinOneDay function could check for null and return false if so? Then it would remove the need for the ternary here.

Reply:
Thanks for pointing that out. It makes a lot more sense to have `wasItemPurchasedWithinLastOneDay` function returning the value of `checked` attribute instead.

Reply:
`wasItemPurchasedWithinLastOneDay` has been modified in the latest commit. https://github.com/the-collab-lab/tcl-19-smart-shopping-list/pull/24/commits/1058294a5fc7d83303651225cb5151d171b4c588